### PR TITLE
10.0 port website canonical url to Odoo 10

### PR DESCRIPTION
**Pull request by zeroheure**
I discovered Renka port after working on this. Anyway this one is complete. Readme updated also.

### CODE REVIEW

**Comment by zeroheure** [comment]
It works. Travis fail because of another module in the tree.

**Review by yajo — CHANGES_REQUESTED** [review]
1. Keep git history: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0#technical-method-to-migrate-a-module-from-90-to-100-branch
2. Fix travis, check [this error](https://travis-ci.org/OCA/website/jobs/337759890#L1329).
3. Thanks! 😊

**Comment by zeroheure** [comment]
@Yajo Thanks. I've updated everything in another repository : zeroheure/website-canonical-url because I don't know how to do it here. I will create a new PR.

**Comment by pedrobaeza** [comment]
@zeroheure you can simply overwrite current branch with that content with `push -f`

**Comment by zeroheure** [comment]
@pedrobaeza Thanks, will try tomorow

File: website_legal_page/views/website_privacy.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
           inherit_id="website.layout_footer_copyright"
```
Comment (simahawk) [inline website_legal_page/views/website_privacy.xml]: files committed by error? :)

File: website_canonical_url/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © initOS GmbH 2016
```
Comment (simahawk) [inline website_canonical_url/models/__init__.py]: (c) -> Copyright

File: website_canonical_url/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © initOS GmbH 2016
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': "Website Canonical URL",
+    'summary': "Canonical URL in Website Headers",
+    'author': "initOS GmbH, Tecnativa, Odoo Community Association (OCA)",
+    'website': "http://www.initos.com",
+    'category': 'Website',
```
Comment (simahawk) [inline website_canonical_url/__manifest__.py]: https://github.com/OCA/website

**Review by simahawk — CHANGES_REQUESTED** [review]
some small things (beside commit history and travis)

**Comment by zeroheure** [comment]
@simahawk Thanks ! done for each file in github.com/zeroheure/website-canonical-url

**Comment by zeroheure** [comment]
NB : I've added absolute url "in a light way" to not increase the server load. See views/template.xml :
`{{request.httprequest.url_root}}..{{canonical_url}}`
it results in URLs like "https://www.chez-les-enfants.fr/../shop" (note the `/../`) but it is valid in html.

**Comment by zeroheure** [comment]
@Yajo The test case don't work any more and I don't know how to update it (I'm not a programmer). Could you help me (you wrote it) ?
Note that canonical url are now absolute while pager urls are relative.

**Comment by zeroheure** [comment]
Please, I need help with code coverity because I don't understand the problem and how to solve it.
I think the requested changes are done.

**Comment by pedrobaeza** [comment]
Code coverage is measuring the quantity of code that is covered against the one is not. In this case, there are less code covered, but I think it can be OK.

**Comment by zeroheure** [comment]
@pedrobaeza Thanks.

**Comment by yajo** [comment]
Hi there, the git history problem is still there.

**Comment by pedrobaeza** [comment]
@Yajo this is 10.0, so no full history is needed to be brought. Is there missing commit history in 9.0 that you have detected?

**Comment by yajo** [comment]
In this case it is needed because this addon was created before the 10.0 branch.

You can check that [none of these](https://github.com/OCA/website/commits/9.0/website_canonical_url) exist. Also 35b33727c128b1ec6908788164589eb9d30fb1f6 creates all files from scratch.

**Comment by zeroheure** [comment]
@Yajo : I though the git history problem was solved (I've followed the docs you have pointed). As it looks difficult, it might be better to open a new PR following the right procedure and close this one ?

**Comment by yajo** [comment]
It's not strictly necessary, you can do whatever changes you need to do, and then `git push --force` to this same branch; then the PR gets updated automatically.

**Comment by simahawk** [comment]
@zeroheure @Yajo what's the status here? I need to migrate to v11 and I would start from here  :)

**Comment by simahawk** [comment]
ah, history missing... I can handle that to speed it up

**Comment by zeroheure** [comment]
@simahawk if you can help me it would be nice. I can't manage to add history.

**Comment by simahawk** [comment]
replaced here https://github.com/OCA/website/pull/449

**Pull request closed without merging**